src/Cool/CoolRequest.php

```diff
@@ -12,169 +12,77 @@
 
 namespace Drupal\collabora_online\Cool;
 
-/**
- * Gets the contents of discovery.xml from the Collabora server.
- *
- * @param string $server
- *   Url of the Collabora Online server.
- *
- * @return string|false
- *   The full contents of discovery.xml, or FALSE on failure.
- */
-function getDiscovery($server) {
-    $discovery_url = $server . '/hosting/discovery';
-
-    $default_config = \Drupal::config('collabora_online.settings');
-    if ($default_config === NULL) {
-        return FALSE;
-    }
-    $disable_checks = (bool) $default_config->get('cool')['disable_cert_check'];
-
-    $stream_context = stream_context_create([
-        'ssl' => [
-            'verify_peer'       => !$disable_checks,
-            'verify_peer_name'  => !$disable_checks,
-        ],
-    ]);
-    $res = file_get_contents($discovery_url, FALSE, $stream_context);
-    return $res;
-}
+use Drupal\collabora_online\Exception\CoolRequestException;
 
 /**
- * Extracts a WOPI url from the parsed discovery.xml.
- *
- * @param \SimpleXMLElement|null|false $discovery_parsed
- *   Parsed contents from discovery.xml from the Collabora server.
- *   Currently, NULL or FALSE are supported too, but lead to NULL return value.
- * @param string $mimetype
- *   MIME type for which to fetch the WOPI url. E.g. 'text/plain'.
- *
- * @return mixed|null
- *   WOPI url as configured for this MIME type in discovery.xml, or NULL if none
- *   was found for the given MIME type.
- */
-function getWopiSrcUrl($discovery_parsed, $mimetype) {
-    if ($discovery_parsed === NULL || $discovery_parsed == FALSE) {
-        return NULL;
-    }
-    $result = $discovery_parsed->xpath(sprintf('/wopi-discovery/net-zone/app[@name=\'%s\']/action', $mimetype));
-    if ($result && count($result) > 0) {
-        return $result[0]['urlsrc'];
-    }
-    return NULL;
-}
-
-/**
- * Checks if a string starts with another string.
- *
- * @param string $s
- *   Haystack.
- * @param string $ss
- *   Needle.
- *
- * @return bool
- *   TRUE if $ss is a prefix of $s.
- *
- * @see str_starts_with()
- */
-function strStartsWith($s, $ss) {
-    $res = strrpos($s, $ss);
-    return !is_bool($res) && $res == 0;
-}
-
-/**
- * Helper class to fetch a WOPI client url.
+ * Service to fetch a WOPI client url.
  */
 class CoolRequest {
```

Comment:
We should rename this service.
However, for now I wanted to keep the name to reduce the disruption in git.
Especially, I did not want to change the name if we are going to change it again later.

Reply:
For now the service has only one responsibility.
But we might want to add more ways to get data from the xml.

Reply:
We agreed on new service names:
- CollaboraDiscovery
- CollaboraDiscoveryFetcher